kvk-api/app.js

```diff
@@ -1,83 +1,126 @@
+  process.env.REDIS_KVK_CACHE_HOST || "localhost"
 );
-redis_kvk_cache.on('connect', function() {
-  console.log('Redis client connected');
+redis_kvk_cache.on("connect", function () {
+  console.log("Redis client connected");
 });
-redis_kvk_cache.on('error', function(error) {
+redis_kvk_cache.on("error", function (error) {
   console.error(error);
 });
 
-const url = process.env.KVK_API_URL || 'api.kvk.nl'
-const endpoint = process.env.KVK_API_ENDPOINT || '/api/v2/testprofile/companies'
-const api_key_name = process.env.KVK_API_KEY
+const url = process.env.KVK_API_URL || "api.kvk.nl";
+const endpoint =
+  process.env.KVK_API_ENDPOINT || "/api/v2/testprofile/companies";
+const api_key_name = process.env.KVK_API_KEY;
 if (api_key_name) {
-  var api_key = null
+  var api_key = null;
   try {
-    api_key = fs.readFileSync('/run/secrets/' + api_key_name, 'utf8');
-    console.log('API key succesfully loaded');
+    api_key = fs.readFileSync("/run/secrets/" + api_key_name, "utf8");
+    console.log("API key succesfully loaded");
   } catch (e) {
-    console.error('API key not found');
+    console.error("API key not found");
   }
 }
 
 async function fetchApi(endpoint, api_key) {
   var headers = {};
   if (api_key) {
     headers = {
-      'apikey': api_key
-    }
+      apikey: api_key,
+    };
   }
-  return fetch('https://' + endpoint, {
-      headers: headers
-    })
-    .catch(err => {
-      console.error('caught', err.message);
+  return fetch("https://" + endpoint, {
+    headers: headers,
+  })
+    .catch((err) => {
+      console.error("caught", err.message);
       return {
-        'error': 1
-      }
+        error: 1,
+      };
     })
-    .then(data => data.json())
+    .then((data) => data.json());
 }
 
 function numberCheck(req, res, next) {
   if (/^\d{8}$/.test(req.params.kvknr)) {
-    next()
+    next();
   } else {
-    res.sendStatus(400)
+    res.sendStatus(400);
   }
 }
 
+function transformData(kvkData) {
+  const {
+    registrationDate,
+    foundationDate,
+    employees,
+    businessActivities,
+    addresses,
+  } = kvkData.data.items[0];
+
+  const data = {
+    registrationDate,
+    foundationDate,
+    employees,
+  };
+
+  // TODO: Check there is exactly one main activity SBI code
+  const mainSbi = businessActivities
+    .filter((activity) => activity.isMainSbi)
+    .map((activity) => activity.sbiCode);
+
+  if (mainSbi.length > 0) {
+    data.mainSbi = mainSbi[0];
+  }
+
+  data.locatedInTheNetherlands =
+    addresses.filter(
+      (address) =>
+        address.type === "vestigingsadres" && address.country === "Nederland"
+    ).length > 0;
+
+  return data;
+}
+
 async function redisFetch(req, res, next) {
-  const kvknr = req.params.kvknr
-  const cacheval = await redis_kvk_cache.get(kvknr)
+  const kvknr = req.params.kvknr;
+  const cacheval = await redis_kvk_cache.get(kvknr);
   if (cacheval) {
-    res.json(JSON.parse(cacheval))
+    res.json(transformData(JSON.parse(cacheval)));
```

Comment:
The data in cache should be already transformed, therefore `transformData` is not needed here

Reply:
See other comment